lib/sles4sap/console_redirection.pm

```diff
@@ -0,0 +1,174 @@
+# SUSE's openQA tests
+#
+# Copyright SUSE LLC
+# SPDX-License-Identifier: FSFAP
+# Maintainer: QE-SAP <qe-sap@suse.de>
+
+package sles4sap::console_redirection;
+use strict;
+use warnings;
+use testapi;
+use Carp qw(croak);
+use Exporter qw(import);
+use serial_terminal qw(select_serial_terminal);
+use Regexp::Common qw(net);
+
+=head1 SYNOPSIS
+
+Library that enables console redirection and file transfers from worker based VM to another host.
+Can be used for cases where worker VM is not the target host for API calls and command execution, but serves only as a jumphost.
+
+=cut
+
+our @EXPORT = qw(
+  connect_target_to_serial
+  disconnect_target_from_serial
+  verify_ip
+  redirection_init
+  check_serial_redirection
+);
+
+my $ssh_opt = '-o StrictHostKeyChecking=no -o ServerAliveInterval=60 -o ServerAliveCountMax=120';
+
+=head2 handle_login_prompt
+
+    handle_login_prompt();
+
+B<ssh_user> Login user
+Detects if login prompt appears and types the password.
+
+=cut
+
+sub handle_login_prompt {
+    my $pwd = get_var('_SECRET_SUT_PASSWORD', $testapi::password);
+    set_serial_term_prompt();
+    my $serial_response = wait_serial(qr/Password:\s*$|:~/i);
+    # Handle password prompt if it appears
+    if (grep /Password:\s*$/, $serial_response) {
+        type_password $pwd;
+        send_key 'ret';
+    }
+    croak 'Failed to setup serial console' unless grep /:~/, $serial_response;
```

Comment:
why `croak` and not `die` ?

Reply:
Could be possible to move the `die ... unless` before the `if(Password:` (Fail-Fast approach) ?

Reply:
I thought the rule is :
library = croak
test module = die

```
Could be possible to move the die ... unless before the if(Password: (Fail-Fast approach) ?
```
The last croak is to check if a prompt appeared after password was typed. But I think I am missing there another 'wait_for_serial'.
I'll double check that logic and come back.

Reply:
I did add the fail fast line as well and moved prompt check into password typing if statement... so the prompt is not checked 2x in case of passwordless login.